### ISSUE TITLE
It's impossible to implement the chains.Chain interface because it depends on a private struct

### ISSUE THREAD

**Issue by kraney** [tmc/langchaingo#1085]
Because ChainCallOption is defined as
```
type ChainCallOption func(*chainCallOption)
```
it's impossible to implement the Chains interface anywhere outside the chains package. You must support these options in the Call() method, but there is no way to read them. To do so, you would have to instantiate a struct that's private to the chains package.

**Comment by pnthang01** [comment]
Exactly, we need to make the struct public so we can make custom LLMChain from it. For now, I have to work around to make it work, that's painful.

**Comment by dnck** [comment]
🤯 Sounds pretty bad. Is someone working on this?
